File: Data Structures & Algorithms/combination-target-sum/submission-0.js
class Solution {
    /**
     * @param {number[]} nums
     * @param {number} target
     * @returns {number[][]}
     */
 combinationSum(nums, target) {
    const result = [];

    function solve(index, currentSum, path) {
        if (currentSum === target) {
            result.push([...path]);  // found a valid combo
            return;
        }
        if (currentSum > target || index === nums.length) {
            return;                  // prune
        }

        // Pick: stay at same index (reuse allowed)
        path.push(nums[index]);
        solve(index, currentSum + nums[index], path);
        path.pop();

        // Skip: move to next index
        solve(index + 1, currentSum, path);
    }

    solve(0, 0, []);
    return result;
}

}

File: Data Structures & Algorithms/best-time-to-buy-and-sell-stock-ii/submission-0.java
class Solution {
    public int maxProfit(int[] prices) {
        int buyp = prices[0];
        int result = 0;
        for (int p : prices) {
            if (p > buyp) {
                result += p - buyp; 
            }
            buyp = p; 
        }
        return result;
    }
}
